From c8151f8bfe6cb002066379d30d0b097c9aff2ca6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sybren=20A=2E=20St=C3=BCvel?= <sybren@blender.org>
Date: Fri, 14 Aug 2026 16:09:19 +0200
Subject: [PATCH] Fix: BKE_image_walk_all_users static cast bugs

`BKE_image_walk_all_users()` visited `mainp->materials` three times,
once as materials, but the other two visits it would cast the pointers
to `Light` and `World`, and call `image_walk_id_all_users()` on them.

The cast is fine in theory, because only the ID data was used, but it
also means any images used by lights & worlds were missed.

Pull Request: https://projects.blender.org/blender/blender/pulls/162671
---
 source/blender/blenkernel/intern/image.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/source/blender/blenkernel/intern/image.cc b/source/blender/blenkernel/intern/image.cc
index 624a5cf9bab9..d1af14bd9bff 100644
--- a/source/blender/blenkernel/intern/image.cc
+++ b/source/blender/blenkernel/intern/image.cc
@@ -3039,13 +3039,13 @@ void BKE_image_walk_all_users(
     image_walk_id_all_users(&ma->id, false, customdata, callback);
   }
 
-  for (Light *light = static_cast<Light *>(mainp->materials.first); light;
+  for (Light *light = static_cast<Light *>(mainp->lights.first); light;
        light = static_cast<Light *>(light->id.next))
   {
     image_walk_id_all_users(&light->id, false, customdata, callback);
   }
 
-  for (World *world = static_cast<World *>(mainp->materials.first); world;
+  for (World *world = static_cast<World *>(mainp->worlds.first); world;
        world = static_cast<World *>(world->id.next))
   {
     image_walk_id_all_users(&world->id, false, customdata, callback);
